sbi/utils/sbiutils.py

```diff
@@ -884,6 +885,64 @@ def within_support(distribution: Any, samples: Tensor) -> Tensor:
         return torch.isfinite(distribution.log_prob(samples))
 
 
+def mcmc_transform(
+    prior: Any,
+    num_prior_samples_for_zscoring: int = 1000,
+    enable_transform: bool = True,
+    device: str = "cpu",
+    **kwargs,
+) -> torch_tf.Transform:
+    """
+    Builds a transform that is applied to parameters during MCMC.
+
+    It does two things:
+    1) When the prior support is bounded, it transforms the parameters into unbounded
+        space.
+    2) It z-scores the parameters such that MCMC is performed in a z-scored space.
+
+    Args:
+        prior: The prior distribution.
+        num_prior_samples_for_zscoring: The number of samples drawn from the prior
+            to infer the `mean` and `stddev` of the prior used for z-scoring. Unused if
+            the prior has bounded support or when the prior has `mean` and `stddev`
+            attributes.
+        enable_transform: Whether or not to use a transformation during MCMC.
+
+    Returns: A transformation that transforms whose `forward()` maps from unconstrained
+        (or z-scored) to constrained (or non-z-scored) space.
+    """
+
+    if enable_transform:
+        if hasattr(prior.support, "base_constraint") and hasattr(
+            prior.support.base_constraint, "upper_bound"
+        ):
+            transform = biject_to(prior.support)
```

Comment:
Sorry, my suggestion in the resolved comment above was not exact:
intuitively the `transform` should transform the parameters from constrained, untransformed to transformed space, and the inverse `transform.inv()` should send transformed parameters back to untransformed space. When it is defined like this it would be the other way around, like in `flows`. 
I would suggest to use `transform = bisect_to(prior.support).inv` to get this logic. do you agree?